src/core/src/op/reduce_logical_and.cpp

```diff
@@ -2,76 +2,61 @@
+
+    template <element::Type_t ET>
+    static result_type visit(const Tensor& in, Tensor& out, const AxisSet& reduction_axes) {
+        using T = fundamental_type_for<ET>;
+        reference::reduce_logical_and(in.data<const T>(), out.data<T>(), in.get_shape(), reduction_axes);
+        return true;
+    }
+};
+}  // namespace reduce_and
 
-op::v1::ReduceLogicalAnd::ReduceLogicalAnd(const Output<Node>& data,
-                                           const Output<Node>& reduction_axes,
-                                           const bool keep_dims)
+namespace v1 {
+ReduceLogicalAnd::ReduceLogicalAnd(const Output<Node>& data, const Output<Node>& reduction_axes, const bool keep_dims)
     : LogicalReductionKeepDims(data, reduction_axes, keep_dims) {
     constructor_validate_and_infer_types();
 }
 
-shared_ptr<Node> op::v1::ReduceLogicalAnd::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> ReduceLogicalAnd::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v1_ReduceLogicalAnd_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<op::v1::ReduceLogicalAnd>(new_args.at(0), new_args.at(1), get_keep_dims());
+    return std::make_shared<ReduceLogicalAnd>(new_args.at(0), new_args.at(1), get_keep_dims());
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace reduce_and {
-namespace {
-bool evaluate_reduce_logical_and(const HostTensorPtr& data,
-                                 const HostTensorPtr& out,
-                                 const AxisSet& reduction_axes,
-                                 bool keep_dims) {
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    out->set_shape(reduce(data->get_shape(), reduction_axes, keep_dims));
-    try {
-        ov::reference::reduce_logical_and(data->get_data_ptr<char>(),
-                                          out->get_data_ptr<char>(),
-                                          data->get_shape(),
-                                          reduction_axes);
-        return true;
-    } catch (const ngraph_error& e) {
-        OPENVINO_WARN << e.what();
-        return false;
-    }
-    OPENVINO_SUPPRESS_DEPRECATED_END
-}
-}  // namespace
-}  // namespace reduce_and
-
-bool op::v1::ReduceLogicalAnd::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool ReduceLogicalAnd::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v1_ReduceLogicalAnd_evaluate);
     OPENVINO_SUPPRESS_DEPRECATED_START
-    OPENVINO_ASSERT(validate_host_tensor_vector(inputs, 2));
-    OPENVINO_ASSERT(validate_host_tensor_vector(outputs, 1));
+    OPENVINO_ASSERT(inputs.size() == 2);
+    OPENVINO_ASSERT(outputs.size() == 1);
 
-    const auto& data = inputs[0];
-    const auto& axes = inputs[1];
-    const auto& out = outputs[0];
-    if (data->get_element_type() != element::boolean || !axes->get_element_type().is_integral_number()) {
```

Comment:
Is `!axes->get_element_type().is_integral_number()` will be handled somewhere in scope of `evaluate`?

Reply:
In others reduction this check is not done in evaluate and `get_normalized_axes_from_tensor` do cast to integral and normalization. The reduction will contain only integral values.
Is also done to more consistent with other reduce operators.